quickstarts/WFLY-15746_WildFly_Application_Development_Guide.adoc

```diff
@@ -0,0 +1,83 @@
+= WildFly Application Development Guide
+:author:            Eduardo Martins
+:email:             emartins@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+This document specifies the development of the Development Guide feature enhancement.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFLY-15746[WFLY-15746]
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/EAP7-1845[EAP7-1845]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:
+
+=== Affected Projects or Components
+
+WildFly (WFLY Documentation and Quickstarts Components)
+
+=== Other Interested Projects
+
+== Requirements
+
+=== Hard Requirements
+
+* The guide should have 2 major sections, one for first time/beginners app developers, and one for advanced app developers.
+
+* The major section for first time app developers should include the following new content:
+** How to develop a first basic application (helloworld quickstart) from scratch using Maven, a complete user story walkthrough, which should include:
+*** How to create the application Maven Project, and add Red Hat Maven repositories and BOMs dependencies
+*** How to package, deploy and run the application (war/bootable jar, web console)
+*** How to add logging to the application (basic usage)
+*** How to test the application
+*** How to use CodeReady Workspaces/Studio (basic usage: add app project, run & debug)
+*** How to build, deploy and run on Openshift (basic usage)
+
+* The major section for first time app developers should also include the following new content:
+** How to develop a first basic application for each specific supported component/technology (e.g. CDI)
+*** Introduction to the component/Technology
+*** Extends the basic application development content (e.g. add needed Maven dependencies related with component/technology)
+*** Any other basic level content which may be useful with respect to the technology
+
+* The following content from the existent Developer Guide should be adapted and included in the Advanced app developers section:
+** Modules and Classloading
+** Logging
+** Specific component/technology content (e.g. CDI)
+
+* The specific content/technology content from the existent Quickstart Guides should be adapted and included in the Advanced app developers section, with respect to the Specific component/technology content.
+
```

Comment:
Will it cover migration like the current developer guide does? https://docs.wildfly.org/29/Developer_Guide.html#How_do_I_migrate_my_application_from_AS7_to_WildFly

Reply:
I think that we should definitely provide high level migration content, and port the EAP migration guide to upstream.